self-hosting/env-variables.mdx

```diff
@@ -0,0 +1,80 @@
+---
+title: Configuration
+description: Complete list of environment variables for LangWatch self-hosting
+---
+
+# Environment Variables
+
+This page provides a comprehensive list of all environment variables that can be configured for LangWatch self-hosting deployments. These variables control various aspects of the application including authentication, database connections, external services, and feature flags.
+
+## Environment Variables Reference
+
+| Variable | Description | Required | Default |
+|----------|-------------|----------|---------|
+| **Core Configuration** |
+| `DATABASE_URL` | PostgreSQL database connection URL | Yes | - |
+| `NODE_ENV` | Application environment (development, test, production) | Yes | - |
+| `BASE_HOST` | Base URL of the application | Yes | - |
+| `API_TOKEN_JWT_SECRET` | Secret key for JWT token signing | Yes | - |
+| **Authentication (NextAuth)** |
+| `NEXTAUTH_PROVIDER` | Authentication provider (azure-ad, okta, auth0, etc.) | No | email |
+| `NEXTAUTH_SECRET` | Secret for NextAuth.js session encryption | Yes | - |
+| `NEXTAUTH_URL` | URL of the authentication service | Yes | - |
+| **Search & Storage** |
+| `ELASTICSEARCH_NODE_URL` | Elasticsearch cluster endpoint | Yes | - |
+| `ELASTICSEARCH_API_KEY` | Elasticsearch API key for authentication | Yes | - |
+| `IS_OPENSEARCH` | Flag to indicate OpenSearch instead of Elasticsearch | No | false |
+| `IS_QUICKWIT` | Flag to indicate Quickwit instead of Elasticsearch | No | false |
+| `REDIS_URL` | Redis connection URL | No | - |
+| `REDIS_CLUSTER_ENDPOINTS` | Redis cluster endpoints (comma-separated) | No | - |
+| **S3 Storage** |
+| `USE_S3_STORAGE` | Enable S3-compatible storage for datasets | No | false |
+| `S3_ENDPOINT` | S3-compatible storage endpoint | No | - |
+| `S3_ACCESS_KEY_ID` | S3 access key ID | No | - |
+| `S3_SECRET_ACCESS_KEY` | S3 secret access key | No | - |
+| `S3_BUCKET_NAME` | S3 bucket name for dataset storage | No | - |
+| `S3_KEY_SALT` | Salt for S3 key encryption | No | - |
+| `DATASET_STORAGE_LOCAL` | Use local storage instead of S3 | No | false |
+| **LLM Providers** |
+| `OPENAI_API_KEY` | OpenAI API key | No | - |
+| `AZURE_OPENAI_ENDPOINT` | Azure OpenAI endpoint URL | No | - |
+| `AZURE_OPENAI_KEY` | Azure OpenAI API key | No | - |
+| `GOOGLE_APPLICATION_CREDENTIALS` | Google Cloud service account credentials | No | - |
+| **NLP Services** |
+| `LANGWATCH_NLP_SERVICE` | Custom NLP service endpoint | Yes | - |
+| `TOPIC_CLUSTERING_SERVICE` | Topic clustering service endpoint | Yes | - |
+| `LANGEVALS_ENDPOINT` | LangEvals evaluation service endpoint | No | - |
+| `LANGWATCH_NLP_LAMBDA_CONFIG` | Lambda configuration for NLP services | No | - |
+| **Security & Encryption** |
+| `CREDENTIALS_SECRET` | Secret for credential encryption, Base16 encoded | No | - |
```

Comment:
why isn't this mandatory, put it high up together with API_TOKEN_JWT_SECRET and the nextauth ones, this way user generated credentials all at once